front/lib/api/assistant/actions/retrieval.ts

```diff
@@ -389,66 +389,31 @@ export async function* runRetrieval(
   );
 
   // Handle data sources list and parents/tags filtering.
-  if (c.dataSources === "all") {
-    const prodCredentials = await prodAPICredentialsForOwner(owner);
-    const api = new DustAPI(prodCredentials);
+  config.DATASOURCE.data_sources = c.dataSources.map((d) => ({
+    workspace_id: d.workspaceId,
+    name: d.name,
+  }));
+
+  for (const ds of c.dataSources) {
+    if (ds.filter.tags) {
+      if (!config.DATASOURCE.filter.tags) {
+        config.DATASOURCE.filter.tags = { in: [], not: [] };
+      }
 
-    const dsRes = await api.getDataSources(prodCredentials.workspaceId);
-    if (dsRes.isErr()) {
-      return yield {
-        type: "retrieval_error",
-        created: Date.now(),
-        configurationId: configuration.sId,
-        messageId: agentMessage.sId,
-        error: {
-          code: "retrieval_data_sources_error",
-          message: `Error retrieving workspace data sources: ${dsRes.error.message}`,
-        },
-      };
+      config.DATASOURCE.filter.tags.in.push(...ds.filter.tags.in);
+      config.DATASOURCE.filter.tags.not.push(...ds.filter.tags.not);
     }
 
-    const ds = dsRes.value.filter((d) => d.assistantDefaultSelected);
-
-    config.DATASOURCE.data_sources = ds.map((d) => {
-      return {
-        workspace_id: prodCredentials.workspaceId,
-        data_source_id: d.name,
-      };
-    });
-  } else {
-    config.DATASOURCE.data_sources = c.dataSources.map((d) => ({
-      workspace_id: d.workspaceId,
-      data_source_id: d.dataSourceId,
```

Comment:
Hum this will break, I need to change it. Adding a commit! 

We had 
```
export type DataSourceConfiguration = {
workspaceId: string;
  dataSourceId: string;
  filter: DataSourceFilter;
};
```

and I changed it for
```
export type DataSourceConfiguration = {
  workspaceId: ModelId;
  name: string;
  filter: DataSourceFilter;
};

because I thought "no foreign" key model ids on types, but I guess we need it!

Reply:
Ok no nevermind it should be good, I'm confused by dataSourceId being the name of the datasource outside of front 😅